astropy/coordinates/earth.py

```diff
@@ -174,6 +176,122 @@ def from_geodetic(cls, lon, lat, height=0., ellipsoid=None):
         self._ellipsoid = ellipsoid
         return self.to(height.unit)
 
+    @classmethod
+    def of_site(cls, site_name):
+        """
+        Return an object of this class for a known observatory/site by name.
+
+        This is intended as a quick convenience function to get basic site
+        information, not a fully-featured exhaustive registry of observatories
+        and all their properties.
+
+        .. note::
+            When this function is called, it will first attempt to
+            download site information from the astropy data server.  If it
+            cannot (i.e., an internet connection is not available), it will fall
+            back on the list included with astropy (which is a limited and dated
+            set of sites).  If you think a site should be added, issue a pull
+            request to the
+            `astropy-data repository <https://github.com/astropy/astropy-data>`_ .
+
+        Parameters
+        ----------
+        site_name : str
+            Name of the observatory (case-insensitive).
+
+        Returns
+        -------
+        site : This class (a `~astropy.coordinates.EarthLocation` or subclass)
+            The location of the observatory.
+
+        See Also
+        --------
+        get_site_names : the list of sites that this function can access
+        """
+        el = cls._get_site_registry()[site_name]
+
+        if cls is el.__class__:
+            return el
+        else:
+            neweel = cls.from_geodetic(*el.to_geodetic())
```

Comment:
Not just any eel: *new* eel.  If only there were an emoji for that...

Will issue a PR shortly.